src/borg/repository.py

```diff
@@ -321,7 +325,10 @@ def get_index_transaction_id(self):
 
     def check_transaction(self):
         index_transaction_id = self.get_index_transaction_id()
-        segments_transaction_id = self.io.get_segments_transaction_id()
+        try:
+            segments_transaction_id = self.io.get_segments_transaction_id()
+        except LoggedIO.AtticSegment:
+            raise self.AtticRepository(self.path)
```

Comment:
Not sure about this one. The whole purpose of this patch is to get rid of the tracebacks and show a short and meaningful error message when running borg commands on Attic repositories...